pep-0432.txt

```diff
@@ -662,47 +668,25 @@ Python data types (not set == ``NULL``) or numeric flags (not set == ``-1``)::
          * be set before calling PyRun_PrepareMain (Py_ReadMainInterpreterConfig
          * will set one of them based on the command line arguments if
          * prepare_main is non-zero when that API is called).
-        int prepare_main;
+        PyBoolObject    *prepare_main;
         PyUnicodeObject *main_source; /* -c switch */
         PyUnicodeObject *main_path;   /* filesystem path */
         PyUnicodeObject *main_module; /* -m switch */
         PyCodeObject    *main_code;   /* Run directly from a code object */
         PyObject        *main_stream; /* Run from stream */
-        int run_implicit_code;        /* Run implicit code during prep */
+        PyBoolObject    *run_implicit_code; /* Run implicit code during prep */
 
         /* Interactive main
          *
          * Note: Settings related to interactive mode are very much in flux.
          */
         PyObject *prompt_stream;      /* Output interactive prompt */
-        int show_banner;              /* -q switch (inverted) */
-        int inspect_main;             /* -i switch, PYTHONINSPECT */
+        PyBoolObject *show_banner;    /* -q switch (inverted) */
+        PyBoolObject *inspect_main;   /* -i switch, PYTHONINSPECT */
 
     } PyMainInterpreterConfig;
 
-
-    /* Struct initialization is pretty horrible in C89. Avoiding this mess would
-     * be the most attractive aspect of using a PyDictObject* instead... */
-    #define _PyArgConfig_INIT  NULL, NULL, NULL, NULL
```

Comment:
Exactly. So between that, and the consistency of access (NULL always = missing, non-NUL always = Python object with the relevant info), we get an API that should be *easier* to work with than a Python dict would be.

I need to explain that more though - I think that's why this was still sitting locally on my HD rather than having pushed live.